src/basic/socket-util.c

```diff
@@ -345,6 +345,31 @@ int sockaddr_port(const struct sockaddr *_sa, unsigned *ret_port) {
         }
 }
 
+union in_addr_union *sockaddr_in_addr(const struct sockaddr *_sa) {
+        union sockaddr_union *sa = (union sockaddr_union*) _sa;
+
+        /* Note that we declare a "const" pointer as input, but return a "non-const" pointer. That way we can
+         * be used both on constant and non-constant socket addresses, and will return an object that is then
+         * itself usable both in constant and non-constant context. Compare with strstr(), which has a
+         * similar logic as it takes a constant string as input, but returns a non-constant one, even though
+         * it is pointing into the same object. */
```

Comment:
Currently, the function is only used in the next commit, and `manager_find_link_address()` takes constant pointer. I prefer to set `const` here, at least now.

Reply:
Adjusted. Force pushed new version. PTAL!